ARKitFullExample/View Controllers/SCNViewController.swift

```diff
@@ -0,0 +1,202 @@
+        
+        // Create a new scene
+        let scene = SCNScene(named: "art.scnassets/ship.scn")!
+        
+        // Set the scene to the view
+        sceneView.scene = scene
+        sceneView.scene.rootNode.scale = SCNVector3(0.2, 0.2, 0.2)
+        
+        /*----👇---- ARVideoKit Configuration ----👇----*/
+        
+        // Initialize ARVideoKit recorder
+        recorder = RecordAR(ARSceneKit: sceneView)
+        // Set the renderer's delegate to retrieve the rendered buffers
+        recorder?.renderAR = self
+        // Enable rendering pre-recording (to push buffers to TwilioVideo)
+        recorder?.onlyRenderWhileRecording = false
+        // Set frames per second rate
+        recorder?.fps = .fps30
+        
+        /*----👇---- TwilioVideo Configuration ----👇----*/
+        
+        self.videoTrack = TVILocalVideoTrack.init(capturer: self)
+        self.audioTrack = TVILocalAudioTrack.init()
+        
+        let options = TVIConnectOptions.init(token: accessToken, block: {(builder: TVIConnectOptionsBuilder) -> Void in
+            if let videoTrack = self.videoTrack {
+                builder.videoTracks = [videoTrack]
+            }
+            if let audioTrack = self.audioTrack {
+                builder.audioTracks = [audioTrack]
+            }
+            builder.roomName = "arkit"
+        })
+        
+        self.room = TwilioVideo.connect(with: options, delegate: self)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        // Create a session configuration
+        let configuration = ARWorldTrackingConfiguration()
+        
+        // Run the view's session
+        sceneView.session.run(configuration)
+        
+        // Prepare the recorder with sessions configuration
+        recorder?.prepare(configuration)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        // Pause the view's session
+        sceneView.session.pause()
+        
+        recorder?.onlyRenderWhileRecording = true
+        recorder?.prepare(ARWorldTrackingConfiguration())
+        
+        // Switch off the orientation lock for UIViewControllers with AR Scenes
+        recorder?.rest()
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    @IBAction func goBack(_ sender: UIButton) {
+        self.dismiss(animated: true, completion: nil)
+    }
+}
+
+//MARK: - ARVideoKit Renderer Method
+extension SCNViewController: RenderARDelegate {
+    func frame(didRender buffer: CVPixelBuffer, with time: CMTime, using rawBuffer: CVPixelBuffer) {
+        if isStreamingVideo {
+            self.frame = TVIVideoFrame(timestamp: Int64(CMTimeGetSeconds(time)),
+                                       buffer: buffer,
+                                       orientation: TVIVideoOrientation.up)
```

Comment:
`TVIVideoFrame` takes timestamp in nano-seconds - [example usage](https://github.com/twilio/video-quickstart-swift/blob/master/ARKitExample/ViewController.swift#L91-L93)